### PR TITLE
[PoC][Auth] Update auth form and add redux

### PR DESCRIPTION
**Pull request by LeonSo7**
[PoC][Auth] Update auth form and add redux

### CODE REVIEW

**Review by TimChoy — APPROVED** [review]

File: src/pages/api/user/register.ts [hunk 1]
```diff
@@ -15,7 +15,9 @@ const registrationHandler = async (
     return;
   }
 
-  const result = registrationSchema.safeParse(req.body);
+  const body = JSON.parse(req.body);
+  const result = registrationSchema.safeParse(body);
+  console.log(result);
```
Comment (arkinmodi) [inline src/pages/api/user/register.ts:20]: Do you need to keep this?

File: src/pages/api/user/register.ts [hunk 2]
```diff
@@ -15,7 +15,9 @@ const registrationHandler = async (
     return;
   }
 
-  const result = registrationSchema.safeParse(req.body);
+  const body = JSON.parse(req.body);
```
Comment (arkinmodi) [inline src/pages/api/user/register.ts:18]: This shouldn't be required. I can send requests through Postman without having to sent the request body as a string.

File: src/pages/auth/login.tsx [hunk 3]
```diff
@@ -1,40 +1,123 @@
...
-        <button>Create Account</button>
-      </Link>
+        <div className={authStyles.authFormCardHeader}>
+          <Icon icon="user" size={80} />
+          <h1>Login</h1>
+        </div>
+        <div className={authStyles.authForm}>
+          <FormGroup label="Email" labelFor="text-input" labelInfo="(Required)">
+            <InputGroup
+              id="text-input"
```
Comment (arkinmodi) [inline src/pages/auth/login.tsx:79]: This shares the same id as the password input. Chrome is giving a warning about it.
```
[DOM] Found 2 elements with non-unique id #text-input: (More info: https://goo.gl/9p2vKq)
<input type="password" id="text-input" placeholder="Password" class="bp4-input" name="password" value="a">
<input type="text" id="text-input" placeholder="Email" class="bp4-input" name="email" value="arkinmodi@gmail.com">
```

**Review by arkinmodi — COMMENTED** [review]
I'm having trouble getting this to work. I can create an account but clicking the login button does nothing. Also there is a lot of commented out code. I get that this is just for the POC but I would prefer not leave a bunch of unused code without comments explaining why they exist and are commented out.

**Pull request merged**